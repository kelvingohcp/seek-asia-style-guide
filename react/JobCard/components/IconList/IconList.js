import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Text from '../../../Text/Text';
import Icon from '../../../Icon/Icon';
import styles from './IconList.less';
import classnames from 'classnames';

const IconList = ({ list, className }) => {
  return (
    <Fragment>
      {
        list.filter(({ show = true }) => show).map(({ iconType, content }, i, filteredList) => (
          <div
            key={i}
            className={classnames({
              [styles.container]: true,
              [styles.last]: filteredList.length === i + 1,
              [className]: className
            })}>
            <div className={styles.iconWrapper}>
              <Icon type={iconType} className={styles.icon} size="small" />
            </div>
            <Text key={i} whispering baseline={false} className={styles.textWrapper}>
              {React.isValidElement(content) ? content : <span>{content}</span>}
            </Text>
          </div>
        ))
      }
    </Fragment>
  );
};

IconList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.shape({
    show: PropTypes.bool,
    iconType: PropTypes.string.isRequired,
    content: PropTypes.oneOfType([PropTypes.string, PropTypes.node])
  })),
  className: PropTypes.string
};
export default IconList;
