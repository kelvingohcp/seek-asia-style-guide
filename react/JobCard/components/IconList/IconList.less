@import (reference) '~seek-asia-style-guide/theme';
@import (reference) '../../JobCard.less';

.container {
  .flexRow();
  padding-bottom: @grid-base;
}

.last {
  padding-bottom: @job-card-section-bottom-space;
}

.textWrapper {
  .text();
  padding-bottom: 0;
}

.iconWrapper {
  display: flex;
  margin-right: @grid-base * 2;
}

.icon {
  color: @saGrey4;
}
