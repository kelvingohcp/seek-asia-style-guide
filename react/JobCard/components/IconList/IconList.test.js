import React from 'react';
import { shallow } from 'enzyme';

import IconList from './IconList';

const LIST = [
  { iconType: 'careerLevel', content: 'Text Only' },
  { iconType: 'careerLevel', content: <div>Text Only</div> }
];

describe('Itemlist - Icon', () => {
  it('should render item list with icon', () => {
    const wrapper = shallow((<IconList list={LIST} />));
    expect(wrapper).toMatchSnapshot();
  });
});
