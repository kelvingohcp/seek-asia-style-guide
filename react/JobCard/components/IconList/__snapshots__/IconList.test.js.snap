// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itemlist - Icon should render item list with icon 1`] = `
<React.Fragment>
  <div
    className="container"
    key="0"
  >
    <div
      className="iconWrapper"
    >
      <Icon
        animation=""
        className="icon"
        lineHeight=""
        rotation=""
        size="small"
        smoothRotate={false}
        type="careerLevel"
      />
    </div>
    <Text
      baseline={false}
      className="textWrapper"
      key="0"
      whispering={true}
    >
      <span>
        Text Only
      </span>
    </Text>
  </div>
  <div
    className="container last"
    key="1"
  >
    <div
      className="iconWrapper"
    >
      <Icon
        animation=""
        className="icon"
        lineHeight=""
        rotation=""
        size="small"
        smoothRotate={false}
        type="careerLevel"
      />
    </div>
    <Text
      baseline={false}
      className="textWrapper"
      key="1"
      whispering={true}
    >
      <div>
        Text Only
      </div>
    </Text>
  </div>
</React.Fragment>
`;
